content/posts/how-to-evolve-application-cicd/index.md

```diff
@@ -0,0 +1,336 @@
++++
+title = "How to evolve an application to make it more robust and maintainable - Setting up the CICD"
+description = ""
+date = 2023-01-15
+featured = false
+draft = false
+comment = true
+toc = true
+reward = true
+pinned = false
+categories = [
+"quality", "test", "CICD"
+]
+tags = [
+""
+]
+series = [
+" How to evolve an application to make it more robust and maintainable"
+]
+images = []
++++
+
+<!--more-->
+
+
+## Foreword
+
+In the last article we detailed the tests added to our project. 
+These tests are useful to check our development locally as we use TDD. But they are also useful to check if there are regressions.
+In this post we will : 
+
+* define what is the CICD
+* How to add the CICD settings in our project
+
+## What is the CICD
+
+The CICD defines Continuous Integration Continuous Delivery. 
+The CICDCD  is Continuous Integration Continuous Delivery Continuous Deployment
+
+When we talk on CICD ( or CICDCD) we often also referred to as the CICD pipelines.
+CICD pipeline defines a process where certain steps are executed one after the other, all with the aim of verifying the application before delivering it and deploying it.
+
+
+## What CICD tools
+
+There are some tools to execute CICD; we can quote:
+* Gitlab-ci
+* CircleCI
+* GitHub Action
+* Jenkins
+
+For open sources projects as our, it's easier to use a CICD tools integrated to a web hosting and software development management service as GitLab or GitHub.
+It's why we'll prefer use GitLab, CircleCi or GitHub Action.
+tools as Jenkins need to deploy another service. 
+
+In this post we will detail these solutions but only two will be implemented: 
+
+* CircleCI
+* GitHub Action
+
+
+## Implementation of our CICD
+
+
+### Our desired pipeline
+
+Through our first pipeline we want to validate :
+
+* compile the code
+* validation of the code formatting
+* the non-regression of the code through the execution of test
+
+
+Also we want the pipeline to be run on every commit on develop branch and ci branch (with prefix ci_).
+We could add other kind of branches : feature(feat_), fixes (fix_), documentation (doc_)
+
+### Implementation with CircleCI
+
+#### First connection
+
+CircleCI is an CICD platform which has a free proposal; 
+To be able to use CircleCI we need : 
+* Create an account
+
+  ![circleci-create-user.png](circleci-create-user.png)
+
+* set up your code
+
+  ![img.png](img.png)
+
+* Create settings
+
+There are tree possibilities to links your project with your future CI:
+
+![circleci-create-settings.png](circleci-create-settings.png)
+
+
+
+#### Creation of job
+
+CircleCI defines "jobs" which are executables tasks; each job can  be divided in steps
+In early version of our CI we want to:
+* build the jar file
+* run the unit tests
+* run the integration tests
+* run the system testThe s
```

Comment:
This sentence is incomplete

Reply:
Done , I removed the end of the sentense